src/gossip/fuzz_table.zig

```diff
@@ -132,31 +133,41 @@ pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
 
                     const should_overwrite = random.boolean();
                     if (should_overwrite) {
-                        logger.debug().logf("overwriting pubkey: {}", .{pubkey});
+                        // logger.debug().logf("overwriting pubkey: {}", .{pubkey});
                         data.wallclockPtr().* = now;
                     } else {
-                        logger.debug().logf("writing old pubkey: {}", .{pubkey});
+                        // logger.debug().logf("writing old pubkey: {}", .{pubkey});
+                        const old_value = random.boolean();
                         const other_insertion_time = insertion_times.items[index];
-                        data.wallclockPtr().* = other_insertion_time -| random.intRangeAtMost(u64, 10, 100);
+                        if (old_value) {
+                            // ignored old value
+                            data.wallclockPtr().* = other_insertion_time -| random.intRangeAtMost(u64, 10, 100);
+                        } else {
+                            // will be a duplicate value
+                            data.wallclockPtr().* = other_insertion_time;
+                        }
                     }
 
                     const signed_data = SignedGossipData.initSigned(&keypair, data);
 
                     // !
                     const result = try gossip_table.insert(signed_data, now);
-                    const did_insert = result.wasInserted();
+                    defer {
+                        if (result == .OverwroteExistingEntry) {
+                            result.OverwroteExistingEntry.deinit(allocator);
```

Comment:
It would be nice if this was supported by the language itself, but I don't think it's worth individually adding an extra method for every variant we need to capture in an if-statement.

Reply:
c959286e